### PR TITLE
Kalman object

### PR DESCRIPTION
**Pull request by emikri**
Some cleaning will be needed

### CODE REVIEW

File: Sensor Fusion/broadcastreceiver/filterloophandler.cpp [hunk 1]
```diff
@@ -2,7 +2,7 @@
 #include <QTimer>
 #include "filterloophandler.h"
 #include "sensorvalues.h"
-
+#include <QFile>
```
Comment (emikri) [inline Sensor Fusion/broadcastreceiver/filterloophandler.cpp:5]: Maybe we should remove the writing to files?
As it is more of an debug/presentation part and bit misplaced where there are? We don't need them anymore?

@zpettzon @alirezatashvir

File: Sensor Fusion/broadcastreceiver/madgwickahrscplusplus.h [hunk 2]
```diff
@@ -15,7 +15,7 @@ class MadgwickAHRScplusplus : public Filter
     QList<float> getOrientation();
     void updateOrientation(float, float, float,float, float, float,float, float, float);
     QQuaternion getRotation();
-
+    void setAngles(QList<float> newAngles);
```
Comment (emikri) [inline Sensor Fusion/broadcastreceiver/madgwickahrscplusplus.h]: Not used?

File: Sensor Fusion/broadcastreceiver/madgwickahrscplusplus.cpp [hunk 3]
```diff
@@ -206,5 +238,13 @@ QList<float> getOrientation() {
 }
 
 QQuaternion MadgwickAHRScplusplus::getRotation(){
-    return QQuaternion(q0, q1, q2, q3);
+    return QQuaternion(q0, -q1, -q2, q3);
+}
+
+void MadgwickAHRScplusplus::setAngles(QList<float> newAngles){
```
Comment (emikri) [inline Sensor Fusion/broadcastreceiver/madgwickahrscplusplus.cpp]: Not used right?

File: Sensor Fusion/broadcastreceiver/kalman.cpp [hunk 4]
```diff
@@ -143,3 +152,151 @@
...
+
+            rad = 0;
+
+        }
+        return rad * 180 / M_PI;
+
+    }
+
+    QList<float> Kalman::getAngleFromVectorDegree(float x, float y, float z, float xMag, float yMag, float zMag){
+
```
Comment (emikri) [inline Sensor Fusion/broadcastreceiver/kalman.cpp:200]: Do we need these comments?

File: Sensor Fusion/broadcastreceiver/madgwickahrscplusplus.cpp (hunk 3 above)
Comment (alirezatashvir) [inline Sensor Fusion/broadcastreceiver/madgwickahrscplusplus.cpp]: +1

File: Sensor Fusion/broadcastreceiver/filterloophandler.cpp (hunk 1 above)
Comment (alirezatashvir) [inline Sensor Fusion/broadcastreceiver/filterloophandler.cpp:5]: I am not sure if we should!

File: Sensor Fusion/broadcastreceiver/kalman.cpp (hunk 4 above)
Comment (alirezatashvir) [inline Sensor Fusion/broadcastreceiver/kalman.cpp:200]: No, we dont need them. +1

File: Sensor Fusion/broadcastreceiver/madgwickahrscplusplus.h (hunk 2 above)
Comment (alirezatashvir) [inline Sensor Fusion/broadcastreceiver/madgwickahrscplusplus.h]: +1

**Pull request merged**